src/xdocs/config_coding.xml

```diff
@@ -5276,6 +5276,25 @@ public class AnnotationLocationCheck extends AbstractCheck {
         <source>
 &lt;module name=&quot;ParameterAssignment&quot;/&gt;
         </source>
+        <p>
+            Example :
+        </p>
+        <source>
+class MyClass {
+  int method(int parameter) {
```

Comment:
This can be made more clear by adding a contract to the input parameter:
```java
 int method(int parameter) {
    if (parameter <= 0 ) {
        throw new IllegalArgumentException("A positive value is expected");
    }
    parameter -= 2;  // violation
    return parameter;
  }	
```
The usual flow of a method is to validate input parameters, then use them. Assigning a value to the parameter is confusing and difficult to read.

Reply:
@ek08 , please do.